tests/test_integration.py

```diff
@@ -72,6 +72,16 @@ def test_generate_json_artifacts(self):
             ],
         )
 
+    def test_generate_json_domain_artifacts(self):
+        self._run_test(
+            'pyexample-json',
+            '_build/json/readthedocs-sphinx-domain-names.json',
+            [
+                'py:exception', 'js:class',
+                '"index.html": "Welcome to pyexample',
```

Comment:
I think we can go more strict here and compare against the full JSON file if it's not incredibly big, or at least a portion of it.

Reading the test it's hard to realize what we are expecting to be the content of that file. From the Python function it seems there is a specific structure containing `types`, `titles` and `paths`.